scipy/sparse/_construct.py

```diff
@@ -361,57 +451,70 @@ def kron(A, B, format=None):
         first matrix of the product
     B : sparse or dense matrix
         second matrix of the product
-    format : str, optional
+    format : str, optional (default: 'bsr' or 'coo')
         format of the result (e.g. "csr")
+        If None, choose 'bsr' for relatively dense array and 'coo' for others
 
     Returns
     -------
-    kronecker product in a sparse matrix format
-
+    kronecker product in a sparse format.
+    Returns a sparse matrix unless either A or B is a
+    sparse array in which case returns a sparse array.
 
     Examples
     --------
     >>> import numpy as np
-    >>> from scipy import sparse
-    >>> A = sparse.csr_matrix(np.array([[0, 2], [5, 0]]))
-    >>> B = sparse.csr_matrix(np.array([[1, 2], [3, 4]]))
-    >>> sparse.kron(A, B).toarray()
+    >>> import scipy as sp
+    >>> A = sp.sparse.csr_array(np.array([[0, 2], [5, 0]]))
+    >>> B = sp.sparse.csr_array(np.array([[1, 2], [3, 4]]))
+    >>> sp.sparse.kron(A, B).toarray()
     array([[ 0,  0,  2,  4],
            [ 0,  0,  6,  8],
            [ 5, 10,  0,  0],
            [15, 20,  0,  0]])
 
-    >>> sparse.kron(A, [[1, 2], [3, 4]]).toarray()
+    >>> sp.sparse.kron(A, [[1, 2], [3, 4]]).toarray()
     array([[ 0,  0,  2,  4],
            [ 0,  0,  6,  8],
            [ 5, 10,  0,  0],
            [15, 20,  0,  0]])
 
     """
-    B = coo_matrix(B)
-
+    # delete next 10 lines and replace _sparse with _array when spmatrix removed
```

Comment:
I'd add the TODO: prefix here to make it more clear that this comment is about future work.